icann-rdap-client/src/gtld/entity.rs

```diff
@@ -0,0 +1,225 @@
+use super::{GtldParams, RoleInfo, ToGtld};
+use icann_rdap_common::contact::PostalAddress;
+use icann_rdap_common::response::entity::Entity;
+
+impl ToGtld for Option<Vec<Entity>> {
+    fn to_gtld(&self, params: &mut GtldParams) -> String {
+        let mut front_formatted_data = String::new();
+        let mut formatted_data = String::new();
+
+        if let Some(entities) = self {
+            for entity in entities {
+                if let Some(roles) = &entity.roles {
+                    for role in roles {
+                        match role.as_str() {
+                            "registrar" => {
+                                if let Some(vcard_array) = &entity.vcard_array {
```

Comment:
I think it would be more future proof if you used the Contact API with from_vcard (https://github.com/icann/icann-rdap/blob/19de4d9628899edf3417b6ecd3a52f0133ceac7a/icann-rdap-common/src/contact/from_vcard.rs#L6) and add any functionality to Contact that is missing.